sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestHelper.scala

```diff
@@ -59,10 +61,39 @@ trait SQLQueryTestHelper extends Logging {
    */
   protected def getNormalizedQueryAnalysisResult(
       session: SparkSession, sql: String): (String, Seq[String]) = {
+    // Note that creating the following DataFrame includes eager execution for commands that create
+    // objects such as views. Therefore any following queries that reference these objects should
+    // find them in the catalog.
     val df = session.sql(sql)
     val schema = df.schema.catalogString
-    // Get the answer, but also get rid of the #1234 expression IDs that show up in analyzer plans.
-    (schema, Seq(replaceNotIncludedMsg(df.queryExecution.analyzed.toString)))
+    val analyzed = df.queryExecution.analyzed
+    // Determine if the analyzed plan contains any nondeterministic expressions.
+    var deterministic = true
+    analyzed.transformAllExpressionsWithSubqueries {
+      case expr: CurrentDate =>
+        deterministic = false
+        expr
+      case expr: CurrentTimestampLike =>
+        deterministic = false
+        expr
+      case expr: CurrentUser =>
+        deterministic = false
+        expr
+      case expr: Literal if expr.dataType == DateType || expr.dataType == TimestampType =>
+        deterministic = false
+        expr
+      case expr if !expr.deterministic =>
+        deterministic = false
+        expr
+    }
+    if (deterministic) {
+      // Perform query analysis, but also get rid of the #1234 expression IDs that show up in the
+      // resolved plans.
+      (schema, Seq(replaceNotIncludedMsg(analyzed.toString)))
+    } else {
+      // The analyzed plan is nondeterministic so elide it from the result to keep tests reliable.
+      (schema, Seq("[Analyzer test output redacted due to nondeterminism]"))
+    }
```

Comment:
Should this be configured? Otherwise every caller who calls this function will start to use non-deterministic codepath?

Reply:
Hi @amaliujia I am a bit confused by your question, would you mind to clarify? This redaction is for the generated analyzer test result files only, in order to prevent the tests from becoming flaky from things like `current_date()` changing each day.

Reply:
I was wondering if this code is used in non testing environment and I tried a code search and it seems only being used with test suites.

Basically I was not sure in production if this part is also used then the `if else` actually changes the output.

At least code search shows this is not a concern.

Reply:
Oh yeah it's only used in the SQL golden test runner. It's in `SQLQueryTestHelper.scala` :)